Link failure on windows-gnu toolchain

Linking to shcore fails when using the windows-gnu toolchain:

`cargo run --example main`
`  = note: c:/mingw/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lshcore`

Commenting out the shcore line in build.rs allow the example to compile and run.
stable-x86_64-pc-windows-gnu 
rustc 1.48.0 (7eac88abb 2020-11-16)

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -2,6 +2,7 @@ environment:
   matrix:
   - TARGET: x86_64-pc-windows-msvc
   - TARGET: i686-pc-windows-msvc
+  - TARGET: x86_64-pc-windows-gnu
 
 platform:
   - x64

--- a/build.rs
+++ b/build.rs
@@ -19,6 +19,8 @@ fn main() {
         println!("cargo:rustc-link-lib=comdlg32");
         println!("cargo:rustc-link-lib=ole32");
         println!("cargo:rustc-link-lib=shell32");
-        println!("cargo:rustc-link-lib=shcore");
+        if !target.ends_with("pc-windows-gnu") {
+            println!("cargo:rustc-link-lib=shcore");
+        }
     }
 }

